runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectMetricsTest.java

```diff
@@ -125,6 +129,74 @@ public void testApplyAttemptedCountersQueryOneNamespace() {
             committedMetricsResult("ns1", "name1", "step2", 0L)));
   }
 
+  private boolean matchesSubPath(String actualScope, String subPath) {
+    return metrics.subPathMatches(actualScope, subPath);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testMatchesSubPath() {
+    // Match of the first element
```

Comment:
Rather than comments, I'd suggest putting this into the assert:

```
assertTrue("Match of the first element",
  matchesSubPath("Top1/Outer1/Inner1/Bottom1", "Top1"));
```

Could also consider putting the actual strings into the assertion message.